homeassistant/components/generic_thermostat/climate.py

```diff
@@ -235,8 +277,14 @@ def _async_startup(event):
                     )
                 else:
                     self._target_temp = float(old_state.attributes[ATTR_TEMPERATURE])
-            if old_state.attributes.get(ATTR_PRESET_MODE) == PRESET_AWAY:
-                self._is_away = True
+            if old_state.attributes.get(ATTR_PRESET_MODE) is None:
+                self._current_preset = self._default_preset
```

Comment:
For maintenance, I'd prefer having the `else` case here. Maybe change this to `is not None` instead and put the fallback case to the `else` part.

As I understand this correct, this is for migration purposes, right? So you can add some kind of comment here to let maintainers know that they can remove this in one of the future versions.

Reply:
This is required to restore the selected preset, after HA reboot, is not for migration purpose. 
This don't introduce BC because it use the same attribute and the same value. 
Cannot be removed in future versions.